pkg/routing/p2p.go

```diff
@@ -116,13 +116,13 @@ func (r *P2PRouter) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	err = r.host.Close()
```

Comment:
Can we add a `<- ctx.Done()` here?

Reply:
I will also note that this causes the host not to get closed cleanly if bootstrapper.Run() returns an error.